ac_dc/oscar_sample_filter.py

```diff
@@ -70,6 +70,42 @@ def load_parameters(lang_oscar_id):
 
 
 class ModifyingSentences:
+    # patterns from https://github.com/joke2k/faker/tree/master/faker/providers/ssn
+    govt_id_regex = {
```

Comment:
I would be very careful on using such regex (ie without using any context surronding the pattern). We should not claim any "cleaning" or "anonymisation" until we have proper evlauation of recision/recall on these.

Reply:
some ids have checksum digits or characters, for these it might be enough to validate the checksum, which could provide good accuracy even without context

Reply:
I think we are not claiming anything except that it matches some version of ids. We are not (as I understand) required legally to do de-identification here. But I think we should attempt to do some. Faker and other libraries do have things like checksum, checking correct number format, etc. And I know these regex are overly broad. I don't think we need to be exact here. We are trying to remove information that *might* otherwise be harmful. If you would like to make it more correct, please! I am sure I goofed something up :)

Reply:
i feel like we should have some analysis of the effect of running these anonymization regexes (as @ggdupont  is suggesting). We need understand if we're not accidentally replacing some non-government ID (or are we okay with false positive?)